readthedocs/core/signals.py

```diff
@@ -57,8 +58,39 @@ def decide_if_cors(sender, request, **kwargs):  # pylint: disable=unused-argumen
         if domain.exists():
             return True
 
+    # Check for Embed API, allowing CORS on public projects
+    # since they are already public
+    if request.path_info.startswith('/api/v2/embed'):
+        url = request.GET.get('url')
+        if url:
+            unresolved = unresolve(url)
+            project = unresolved.project
+            version_slug = unresolved.version_slug
+        else:
+            project_slug = request.GET.get('project', None)
+            version_slug = request.GET.get('version', None)
+            project = Project.objects.filter(slug=project_slug).first()
+
+        if project and version_slug:
+            # This is from IsAuthorizedToViewVersion,
+            # we should abstract is a bit perhaps?
```

Comment:
Can you explain this a bit more? I didn't see how to integrate CORS into the view.

Reply:
You'll need to set the header manually (this header https://github.com/zestedesavoir/django-cors-middleware/blob/50d07ca143ea230e9d046637a0ace78ebdcfbc02/corsheaders/middleware.py#L125) and any other headers you think we should add. Was reading the code from that package to see what headers they put, and realized two things:

- The project is archived
- The is_enabled and check_signal methods are called several times, in our case we're doing some db queries on the signal, so may be repeated like 3 or 4 times...

Reply:
I did this for the cache tags https://github.com/readthedocs/readthedocs.org/blob/91e1530dd09676352a00d9b1f0c053fa8bf660cc/readthedocs/api/v2/mixins.py#L19-L24

Reply:
The header is sent on an OPTIONS request, and I don't believe will actually hit the view will it?

Reply:
> The is_enabled and check_signal methods are called several times, in our case we're doing some db queries on the signal, so may be repeated like 3 or 4 times...

These won't be hit very frequently I don't believe, so probably safe?

Reply:
https://stackoverflow.com/questions/18181261/handle-options-request-with-djangos-built-in-server seems to say that views by default won't respond to `OPTIONS`, so that's part of what the middleware is doing I think.

Reply:
Nope, is set in the get response https://github.com/zestedesavoir/django-cors-middleware/blob/50d07ca143ea230e9d046637a0ace78ebdcfbc02/corsheaders/middleware.py#L122-L126

the options check is to add more info headers.

Reply:
The OPTIONS check is sent before the `GET` request is sent, and is what allows the `GET` to progress: https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#preflighted_requests

Reply:
> These won't be hit very frequently I don't believe, so probably safe?

I mean, we are already executing those queries several times for the api, but yeah, don't think those APIs are hit that much.

Reply:
> The OPTIONS check is sent before the GET request is sent, and is what allows the GET to progress: https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#preflighted_requests 

That says that simple requests skip the pre-flight https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#simple_requests

Get is listed as a simple request, I just checked locally and I don't see any options request being sent.

Reply:
Hrm, odd. Definitely something to research more.